lib/Migration/Version005000001Date20251124093628.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\ContextChat\Migration;
+
+use Closure;
+use OCA\ContextChat\AppInfo\Application;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use Override;
+
+class Version005000001Date20251124093628 extends SimpleMigrationStep {
+	private static array $configKeys = [
+		'action_job_interval',
+		'auto_indexing',
+		'backend_init',
+		'crawl_job_interval',
+		'fs_listener_job_interval',
+		'indexed_files_count',
+		'indexing_batch_size',
+		'indexing_job_interval',
+		'indexing_max_size',
+		'indexing_max_time',
+		'installed_time',
+		'last_indexed_file_id',
+		'last_indexed_time',
+		'logfile',
+		'providers',
+	];
+
+	public function __construct(
+		private IDBConnection $connection,
+	) {
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	#[Override]
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+		$schema = $schemaClosure();
+
+		if (!$schema->hasTable('appconfig')) {
+			return null;
+		}
+
+		$qb = $this->connection->getQueryBuilder();
+		$qb->update('appconfig')
```

Comment:
Mmh, why are you bypassing the Appconfig interface? The db table is not guaranteed to be stable, while the OCP interface is

Reply:
Mh, that's a conundrum :thinking: 

We could use https://nextcloud-server.netlify.app/classes/ocp-iappconfig#method_getKeys to get the known config keys, perhaps?